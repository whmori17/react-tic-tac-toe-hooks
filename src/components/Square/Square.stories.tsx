import * as React from 'react';
import { Square } from '.';

export default {
  title: 'Square',
};

const onClick = (i: number) => console.log(i);

export const emptySquare = () => <Square value={''} {...{ onClick }} />;
export const xSquare = () => <Square value={'X'} {...{ onClick }} />;
export const circleSquare = () => <Square value={'O'} {...{ onClick }} />;
