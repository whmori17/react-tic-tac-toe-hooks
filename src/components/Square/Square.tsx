import React from 'react';

export interface SquareProps {
  onClick: (i: number) => void;
  value: string;
}

export const Square: React.FC<SquareProps> = ({ value, onClick }) => {
  return (
    <button className={'square'} {...onClick}>
      {value}
    </button>
  );
};
