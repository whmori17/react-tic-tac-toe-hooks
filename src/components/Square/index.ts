export { Square, SquareProps } from './Square';
