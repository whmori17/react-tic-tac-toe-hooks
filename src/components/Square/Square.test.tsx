import * as React from 'react';
import { shallow } from 'enzyme';
import { Square, SquareProps } from './Square';

describe('Square', () => {
  it('renders', () => {
    const squareProps: SquareProps = {
      value: 'X',
      onClick: (i: number) => console.log(i),
    };
    const wrapper = shallow(<Square {...squareProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
