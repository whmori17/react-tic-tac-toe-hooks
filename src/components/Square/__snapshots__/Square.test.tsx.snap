// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Square renders 1`] = `
<button
  className="square"
>
  X
</button>
`;
